File: bms/go/geo/geo_r1_Interval_ApproxEqual/geo_r1_Interval_ApproxEqual.go
package test

import (
	"math"
)

var ()

const (
	geo_s2_epsilon = 1e-15
	geo_r1_epsilon = 1e-15
)

func (i geo_r1_Interval) ApproxEqual(other geo_r1_Interval) bool {
	if i.IsEmpty() {
		return other.Length() <= 2*geo_r1_epsilon
	}
	if other.IsEmpty() {
		return i.Length() <= 2*geo_r1_epsilon
	}
	return math.Abs(other.Lo-i.Lo) <= geo_r1_epsilon &&
		math.Abs(other.Hi-i.Hi) <= geo_r1_epsilon
}

func (i geo_r1_Interval) IsEmpty() bool { return i.Lo > i.Hi }

func (i geo_r1_Interval) Length() float64 { return i.Hi - i.Lo }

type geo_r1_Interval struct {
	Lo, Hi float64
}

File: translations_only/success/go/geo/geo_r1_Interval_ApproxEqual/claude2/Hinted/geo_r1_Interval_ApproxEqual.rs

use std::f64::consts::{E, PI};

const GEO_S2_EPSILON: f64 = 1e-15;
const GEO_R1_EPSILON: f64 = 1e-15;

struct GeoR1Interval {
    lo: f64,
    hi: f64,
}

fn approx_equal(i: &GeoR1Interval, other: &GeoR1Interval) -> bool {
    if is_empty(i) {
        return length(other) <= 2.0 * GEO_R1_EPSILON;
    }
    if is_empty(other) {
        return length(i) <= 2.0 * GEO_R1_EPSILON;
    }
    (other.lo - i.lo).abs() <= GEO_R1_EPSILON
        && (other.hi - i.hi).abs() <= GEO_R1_EPSILON
}

fn is_empty(i: &GeoR1Interval) -> bool {
    i.lo > i.hi
}

fn length(i: &GeoR1Interval) -> f64 {
    i.hi - i.lo
}

